controllers/metal3machine_controller.go

```diff
@@ -365,6 +404,24 @@ func (r *Metal3MachineReconciler) BareMetalHostToMetal3Machines(obj handler.MapO
 	return []ctrl.Request{}
 }
 
+// Metal3DataToMetal3Machines will return a reconcile request for a Metal3Machine if the event is for a
+// Metal3Data and that Metal3Data references a Metal3Machine.
+func (r *Metal3MachineReconciler) Metal3DataToMetal3Machines(obj handler.MapObject) []ctrl.Request {
+	if m3d, ok := obj.Object.(*capm3.Metal3Data); ok {
+		if m3d.Spec.Metal3Machine != nil {
+			return []ctrl.Request{
+				ctrl.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      m3d.Spec.Metal3Machine.Name,
+						Namespace: m3d.Namespace,
```

Comment:
why not checking if the m3d.Spec.Metal3Machine.Namespace exists or not?

Reply:
done, thanks